google/cloud/forseti/scanner/audit/kms_rules_engine.py

```diff
@@ -346,34 +338,107 @@ def __init__(self, rule_name, rule_index, rule):
         self.rule_index = rule_index
         self.rule = rule
 
-    def is_key_rotated(self, creation_time, scan_time):
-        """Check if the key has been rotated within the time speciifed in the
-         policy.
+    def find_match_rotation_period(self, key, rotation_period, mode):
+        """Check if there is a match for this rule rotation period against the
+        given resource.
 
         Args:
-            creation_time (datetime): The time at which the primary version of
-            the key was created.
-            scan_time (datetime): Snapshot timestamp.
+            key (Resource): The resource to check for a match.
+            mode (string): The mode specified in the rule.
+            rotation_period (string): The cut off rotation schedule of crypto
+            key specified in rule file.
 
         Returns:
-            bool: Returns true if key was rotated within the time specified.
+            bool: Returns true if a match is found.
```

Comment:
is a match a violation ?

Reply:
If the mode is `blacklist` and match is found or if the mode is `whitelist` and there's no match then it is a violation.